src/scheduler/fifo.c

```diff
@@ -1229,33 +1228,32 @@ run_job(int pbs_sd, resource_resv *rjob, char *execvnode, int throughput, schd_e
 	}
 
 	if (!rc) {
-		if (pbs_sd != SIMULATE_SD) {
-			if (rjob->is_shrink_to_fit) {
-				char timebuf[TIMEBUF_SIZE] = {0};
-				rc = 1;
-				/* The job is set to run, update it's walltime only if it is not a foerever job */
-				if (rjob->duration != JOB_INFINITY) {
-					convert_duration_to_str(rjob->duration, timebuf, TIMEBUF_SIZE);
-					rc = update_job_attr(pbs_sd, rjob, ATTR_l, "walltime", timebuf, NULL, UPDATE_NOW);
-				}
-				if (rc > 0) {
-					if (strlen(timebuf) > 0) {
-						char logbuf[MAX_LOG_SIZE];
-						snprintf(logbuf, MAX_LOG_SIZE, "Job will run for duration=%s", timebuf);
-						schdlog(PBSEVENT_SCHED, PBS_EVENTCLASS_JOB, LOG_NOTICE, rjob->name, logbuf);
-					}
-					if (throughput)
-						rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
-					else
-						rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
+		if ((pbs_sd == SIMULATE_SD) ||
```

Comment:
If we do not re-order then we need the following block of code as well to work for simulation case.
if (pbs_sd == SIMULATE_SD) {
		if (throughput)
				rc = pbs_asyrunjob(pbs_sd, rjob->name, execvnode, NULL);
			else
				rc = pbs_runjob(pbs_sd, rjob->name, execvnode, NULL);
}
Instead of adding the above additional block, I reordered  the above existing if/else block itself so that it works for both simulation and non-simulation case.